crypto/hmac_sha256.go

```diff
@@ -0,0 +1,97 @@
+package crypto
+
+import (
+	"context"
+	"crypto/ed25519"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+	"github.com/mr-tron/base58"
+	"golang.org/x/crypto/blake2b"
+)
+
+// HmacSha256WithPrk will HmacSha256 using the provided prk.  See HmacSha256 for
+// options supported.
+func HmacSha256WithPrk(ctx context.Context, data, prk []byte, opt ...wrapping.Option) (string, error) {
+	opt = append(opt, WithPrk(prk))
+	return HmacSha256(ctx, data, nil, nil, nil, opt...)
+}
+
+// HmacSha256 the provided data. Supports WithPrefix, WithEd25519 and WithPrk
+// options. WithEd25519 is a "legacy" way to complete this operation and should
```

Comment:
I wonder if you should just not even document WithEd25519 and put a deprecation notice on the option. Or call it something like WithDeprecatedEd25519. Arguably we shouldn't even support it at all -- it feels weird to add new code that is already deprecated.

Reply:
Well... Boundary uses it, so if we don't add it then boundary won't be able to use this.

Reply:
happy to mark it deprecated though.

Reply:
🤔 about this a bit more... will all ed25519 options be deprecated going forward?  I think, it's just this particular op that it's deprecated.